pennylane/math/decomposition.py

```diff
@@ -0,0 +1,226 @@
+    :math:`e^{i\alpha} RX(\phi) RZ(\theta) RX(\lambda)`.
+
+    Args:
+        U (array): 2x2 unitary matrix
+        return_global_phase (bool): if True, returns the global phase as well.
+
+    Returns:
+        tuple: The rotation angles :math:`\lambda`, :math:`\theta`, and :math:`\phi` and the
+            global phase :math:`\alpha` if ``return_global_phase=True``.
+
+    """
+
+    U, global_phase = math.convert_to_su2(U, return_global_phase=True)
+    EPS = math.finfo(U.dtype).eps
+
+    # Compute \phi, \theta and \lambda after analytically solving for them from
+    # U = RX(\phi) RZ(\theta) RX(\lambda)
+    sum_diagonal_real = math.real(U[..., 0, 0] + U[..., 1, 1])
+    sum_off_diagonal_imag = math.imag(U[..., 0, 1] + U[..., 1, 0])
+    half_phi_plus_lambdas = math.arctan2(-sum_off_diagonal_imag, sum_diagonal_real + EPS)
+    diff_diagonal_imag = math.imag(U[..., 0, 0] - U[..., 1, 1])
+    diff_off_diagonal_real = math.real(U[..., 0, 1] - U[..., 1, 0])
+    half_phi_minus_lambdas = math.arctan2(diff_off_diagonal_real, -diff_diagonal_imag + EPS)
+    lam = half_phi_plus_lambdas - half_phi_minus_lambdas
+    phi = half_phi_plus_lambdas + half_phi_minus_lambdas
+
+    # Compute \theta
+    theta = math.where(
+        math.isclose(math.sin(half_phi_plus_lambdas), math.zeros_like(half_phi_plus_lambdas)),
+        2
+        * math.arccos(
+            math.clip(sum_diagonal_real / (2 * math.cos(half_phi_plus_lambdas) + EPS), -1, 1)
+        ),
+        2
+        * math.arccos(
+            math.clip(-sum_off_diagonal_imag / (2 * math.sin(half_phi_plus_lambdas) + EPS), -1, 1)
+        ),
+    )
+
+    phi = math.squeeze(phi % (4 * np.pi))
+    theta = math.squeeze(theta % (4 * np.pi))
+    lam = math.squeeze(lam % (4 * np.pi))
+
+    return (lam, theta, phi, global_phase) if return_global_phase else (lam, theta, phi)
+
+
+def zxz_rotation_angles(U, return_global_phase=False):
+    r"""Compute the rotation angles :math:`\lambda`, :math:`\theta`, and :math:`\phi` and the
+    phase :math:`\alpha` of a 2x2 unitary matrix as a product of Z and X rotations in the form
+    :math:`e^{i\alpha} RZ(\phi) RX(\theta) RZ(\lambda)`.
+
+    Args:
+        U (array): 2x2 unitary matrix
+        return_global_phase (bool): if True, returns the global phase as well.
+
+    Returns:
+        tuple: The rotation angles :math:`\lambda`, :math:`\theta`, and :math:`\phi` and the
+            global phase :math:`\alpha` if ``return_global_phase=True``.
+
+    """
+
+    U, global_phase = math.convert_to_su2(U, return_global_phase=True)
+    EPS = math.finfo(U.dtype).eps
+
+    abs_a = math.clip(math.abs(U[..., 0, 0]), 0, 1)
+    abs_b = math.clip(math.abs(U[..., 0, 1]), 0, 1)
+    theta = math.where(abs_a > abs_b, 2 * math.arccos(abs_a), 2 * math.arcsin(abs_b))
+
+    half_phi_plus_lam = math.angle(U[..., 1, 1] + EPS)
+    half_phi_minus_lam = math.angle(1j * U[..., 1, 0] + EPS)
```

Comment:
also why they look so different from previous `zxz`? I don't see previous version being using U[1,1] and U[1,0]. Are they somehow eventually equivalent?